Add QtCreator 4.13 and 4.14

dimitrido opened the pull request:
Based on pull request #19 , I added QtCreator 4.13 and 4.14 CIs for Linux, Mac and Windows.

dimitridout committed:
Updated README.md

dimitridout committed:
Enable build for QtCreator 4.13 and .14 for mac&linux

dimitridout committed:
Enable build for QtCreator 4.13 and .14 for windows

dimitridout committed:
Adapt to new download URL for QtCreator 5.13 and above

danya140 commented on .travis.yml:
```diff
@@ -42,6 +42,16 @@ matrix:
+          compiler: gcc
+          sudo: required
+          dist: trusty
+          env: QTC_VER=4.14 QTC_DL=linux_gcc_64_rhel72
```
Looks like from 4.13 qt Linux download named linux_x64 not linux_gcc_64_rhel72

dimitrido committed:
Changed Qt download name to linux_x64

Commit:
Add .7z to ignore
Add QT patch version to environment variable to be able build not only 0 patch. Cause for some reason 4.14.0 doesnt have any .pri
Update code to meet version 4.14
Fix problems due to changes in path structure in qtcreator-dev. In 4.14 src moved to include/src

danya140 commented:
@dimitrido, I'm able to fix issues with build on 4.14, but can't add commit directly to your pull request, so I created pull request to your fork. Please review it and consider add those fixes here

dimitrido committed:
Merge pull request #1 from danya140/master

Fixes for 4.14

dimitrido commented:
Thanks @danya140 , I just merged it

danya140 commented on qtc-sourcetrail.pro:
```diff
@@ -63,7 +65,14 @@ QTC_PLUGIN_RECOMMENDS += \
+    include($$IDE_SOURCE_TREE/include/src/qtcreatorplugin.pri)
+}
+
+lessThan(QTC_VER_MINOR, 13) | lessThan(QTC_VER_MAJOR, 4){
```
Oops. Misspelled. Must be less than 14

danya140 commented:
Also, it looks like from 4.14 qt decided to use c++17 features, so someone from maintainers must update build machines or do something else, cause gcc 4.8.4 doesn't support these features.
Compare src/libs/utils/optional.h from 4.14 and 4.13 for more info
